frontend/src/components/forms/TextInput.tsx

```diff
@@ -34,15 +46,51 @@ const TextInput: FunctionComponent<Props> = ({
     },
   )
 
-  const registerOptions: RegisterOptions = {
-    required: otherProps.required,
-    valueAsNumber: type === 'number',
+  if (register != null) {
```

Comment:
suggestion (non-blocking): change to `if (register)` for a much smaller early return block